rosbag2_storage_mcap/src/mcap_storage.cpp

```diff
@@ -136,12 +215,18 @@ MCAPStorage::~MCAPStorage() {
 #ifdef ROSBAG2_STORAGE_MCAP_HAS_STORAGE_OPTIONS
```

Comment:
is this ifdef part of the plugin API? is this something we plan to clean up at some point?

Reply:
It's specific to `rosbag2_storage_mcap` - this is part of the plugin's strategy to build against all live ROS 2 distros. The writer API changed between Foxy and present (switching from `string uri` to `StorageOptions options` args), the ifdef is being used to switch between those API versions.

See https://github.com/ros-tooling/rosbag2_storage_mcap/blob/main/rosbag2_storage_mcap/CMakeLists.txt#L46

We can clean up this particular one when Foxy reaches EOL (May 2023)